rpadmin/api_migration.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package rpadmin
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+)
+
+const (
+	baseMigrationEndpoint = "/v1/migrations/"
+)
+
+// AddMigration adds a migration to the cluster. It accepts one of InboundMigration or OutboundMigration.
+func (a *AdminAPI) AddMigration(ctx context.Context, migration any) (int, error) {
+	migrationType := reflect.TypeOf(migration)
```

Comment:
To avoid having `any` in the input, what do you think of splitting the `AddMigration` to `AddInboudMigration` and `AddOutbountMigration`. It's easier for the caller to figure the usage.